Wordle.Api/Wordle.Api/Services/LeaderboardService.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+using Wordle.Api.Data;
+
+namespace Wordle.Api.Services
+{
+    public class LeaderboardService
+    {
+        private readonly AppDbContext _db;
+
+        public LeaderboardService(AppDbContext db)
+        {
+            _db = db;
+        }
+
+        public async Task<IEnumerable<Player>> GetTopTenScores()
+        {
+            return await _db.Players.OrderByDescending(player => player.AverageAttempts).Take(10).ToListAsync();
+        }
+
+        public async Task<Player> AddNewScore(string? name, double attempts)
+        {
+            // Used null forgiveness because compiler is not recognizing
+            // that null is impossible with the condition
+            string playerName = name.IsNullOrEmpty() ? "Guest" : name!;
+            if(_db.Players.Any(p => p.Name == playerName))
+            {
```

Comment:
by any chance could you at some point make it would remember when I come back to the website that It would remember my name? Just a thought

Reply:
Fixed, thanks